File: 3rdSemester/Object Oriented Programming Lab/Lab 1/Python/Activity1_4_1.py
class heater:
    def __init__(self):
        self.temp = 15

    def warmer(self):
        self.temp += 5

    def cooler(self):
        self.temp -= 5

    def show(self):
        return self.temp

h1 = heater()
h2 = heater()

h1.cooler()
print(h1.show())
print("For Heater 1 ")
print("Temperature: ")
print(h1.show())
h1.warmer()
print("Warmer Temperature: ")
print(h1.show())
h1.cooler()
print("Cooler Temperature: ")
print(h1.show())
print("For Heater 2 ")
print("Temperature: ")
print(h2.show())
h2.warmer()
h2.warmer()
print("Warmer Temperature: ")
print(h2.show())
h2.cooler()
print("Cooler Temperature: ")
print(h2.show())

File: 3rdSemester/Object Oriented Programming Lab/Lab 1/Activity1_4_1.cpp
#include <iostream>
using namespace std;

class Heater
{

private:
    int temperature;

public:
    Heater()
    {
        temperature = 15;
    }

    void warmer()
    {
        temperature += 5;
    }

    void cooler()
    {
        temperature -= 5;
    }

    int access_Temperature()
    {
        return temperature;
    }
};

int main()
{
    Heater h1,h2;

    system("cls");
    cout<<"Temperature = "<< h1.access_Temperature()<<endl;
  
    h1.warmer();
    cout<<"Temperature after warmer= "<< h1.access_Temperature()<<endl;  

    h1.cooler();
    cout<<"Temperature after cooler= "<< h1.access_Temperature()<<endl;

    cout<<"Temperature = "<< h1.access_Temperature()<<endl;
  
    h2.warmer();
    cout<<"Temperature after warmer= "<< h2.access_Temperature()<<endl;  

    h2.cooler();
    cout<<"Temperature after cooler= "<< h2.access_Temperature()<<endl;

    h2.cooler();
    cout<<"Temperature after again cooler= "<< h2.access_Temperature()<<endl;

    system("pause");
    return 0;
}
